src/polynomial/poly-grads.jl

```diff
@@ -0,0 +1,386 @@
+import ChainRulesCore: frule, rrule
+using LinearAlgebra
+
+function compute_gn_jac(u_n::AbstractVector{T}) where T
+    n = length(u_n) - 1
+    g_n = zero(u_n)
+    dgdu = fill!(similar(u_n, n + 1, n + 1), zero(T))
+
+    ## First: calculate the a_n terms (eqn. 10)
+    a_n = zero(g_n)
+    dadu = copy(dgdu)
+    # constant term
+    a_n[begin] = one(T)
+    for i in 1:n
+        bcoeff = one(T) # binomial coefficient
+        for j in 0:i
+            r = (-1)^j * bcoeff
+            a_n[begin + j] -= r * u_n[begin + i]
+            dadu[begin + i, begin + j] -= r
+            bcoeff *= (i - j) / (j + 1)
+        end
+    end
+
+    ## Second: calculate the g_n terms (eqn. 15)
+    for j in n:-1:2
+        if j ≥ n - 1
+            g_n[begin + j] = a_n[begin + j] / (j + 2)
+            for i in 0:n - 1
+                dgdu[begin + i + 1, begin + j] = dadu[begin + i + 1, begin + j] / (j + 2)
```

Comment:
Isn't it equivalent to
```suggestion
            for i in 1:n
                dgdu[begin + i, begin + j] = dadu[begin + i, begin + j] / (j + 2)
```